### DISCUSSION TITLE
Filters for external datasource

### DISCUSSION THREAD

**Discussion by ars3nicum** [Budibase/budibase#9988] [Help] [answered]
I am using Postgres database as a datasource and add a dynamic filters to a table.
Everything works well until I want to filter some values comparing with 0 (zero), it doesn't do anything.
For internal database it is working normal.
Is it an issue or I am doing something wrong?

**Comment by ConorWebb96** [comment]
Hey @ars3nicum,

Without seeing some example screenshots of the data you are filtering, as well as the dynamic filter parameters you are using. It will be quite hard to see the issues.

Could you provide some additional information so we can have a look.

Thanks!

**Comment by ars3nicum** [comment]
I have numbers in table (integers and floats) and if I add filter for that fields I can't filter more than /less than/equal to 0.

<img width="283" alt="Screenshot 2023-03-13 at 06 50 18" src="https://user-images.githubusercontent.com/90869728/224667027-3cfb8422-4d49-4f69-a745-68c29c8d3d78.png">
<img width="990" alt="Screenshot 2023-03-13 at 06 52 24" src="https://user-images.githubusercontent.com/90869728/224667053-c39732ab-31c8-48eb-ba2c-64b8dd5d0966.png">

**Comment by ars3nicum** [comment]
Any chances to look into it? I am stuck.

**Reply by ConorWebb96** [reply]
Hey @ars3nicum, 

Sorry for the long reply. I haven't been able to recreate this, the dynamic filter seems to work for me. 
Could you send me across your app export?

Thanks.

**Reply by ars3nicum** [reply]
Hi. I am attaching example app export and an SQL query to create a test table which to use as a datasource:
I use PostgreSQL.

```
CREATE TABLE test(  
	id int NOT NULL PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
    param1 numeric,
    param2 numeric
);

INSERT INTO test (param1, param2) VALUES (1,2);
INSERT INTO test (param1, param2) VALUES (2,3);
INSERT INTO test (param1, param2) VALUES (-1,5);
INSERT INTO test (param1, param2) VALUES (5,-1);
```

Try to add filter where `param1` is more than or equal to 0.

[Test-export-1679071901716.tar.gz](https://github.com/Budibase/budibase/files/11003860/Test-export-1679071901716.tar.gz)

**Reply by ars3nicum** [reply]
I am attaching one more app export with added budibase inner base with the same data, so you can compere behaviour of filters side by side.
[Test-export-1679085972534.tar.gz](https://github.com/Budibase/budibase/files/11006144/Test-export-1679085972534.tar.gz)

**Comment by ars3nicum** [comment]
so any update on this?

**Comment by ars3nicum** [comment]
Please look into it.

**Comment by ConorWebb96** [comment]
Hey @ars3nicum,

I've had a more in-depth look at this and you are right you can't dynamically filter on negative numbers on postgress/other data sources that aren't the Budibase tables. I think there may be a bug with the dynamic filtering with regards to negative numbers. I'll confirm this on Monday with the engineering team and go from there to see what we can do to get this fixed.

For now my advice would be to use custom queries in order to do this.  Theres a lot of information within our docs which may aid you with this.

The query you'd need would be something similar to the below.

`SELECT * FROM test WHERE param1 <= 0;` you can customise this to make it dynamic e.g. the rule being applied as well as wha the value its been checked against.

I hope this helps, and my apologies for the long response time.

**Comment by ars3nicum** [comment]
Thank you!

**Comment by ConorWebb96** [comment]
Hey @ars3nicum,

Could you try filtering by 0.1 instead of 0 and see if this works? This is likely related to a falsey check where 0 is being treated as a falsey value, therefore skipping the whole condition entirely and not filtering anything.

I hope this helps!

**Comment by ars3nicum** [comment]
Hi. Non-zero values works well. I agree that 0 is treated as null here.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @ars3nicum,

Firstly, thank you for bringing this issue to light.

I've created an issue here: https://github.com/Budibase/budibase/issues/10118 which should hopefully be fixed. I don't think it should be a very difficult one to fix.

Hopefully, the 2 work arounds will help you in the meantime!
